Framework/Algorithms/src/ConvertToConstantL2.cpp

```diff
@@ -104,7 +114,9 @@ void ConvertToConstantL2::exec() {
     oldPos.getSpherical(r, theta, phi);
     V3D newPos;
     newPos.spherical(m_l2, theta, phi);
-    outputDetInfo.setPosition(i, newPos);
+
+    const size_t detIndex = inputSpecInfo.detector(i).index();
```

Comment:
@PranavBahuguna It is in a branch that was merged into master last week. If your changes are based on an older version, you may need to merge in master first.